xarray/tests/test_distributed.py

```diff
@@ -1,23 +1,29 @@
 """ isort:skip_file """
-
+from __future__ import absolute_import, division, print_function
+import os
 import sys
+import pickle
+import tempfile
 
 import pytest
 
 dask = pytest.importorskip('dask')  # isort:skip
 distributed = pytest.importorskip('distributed')  # isort:skip
 
 from dask import array
+from dask.distributed import Client, Lock
```

Comment:
@mrocklin - would you mind looking at the test implementation I have here and let us know if you see anything that would be causing the default (global) dask scheduler to be permanently overridden. In #1971, I pointed to some test failures that appear coming from a unexpected scheduler.

Reply:
Everything seems fine to me.  What is `dask._config.globals['get']`?  Is it the `get` method of a client?  To debug you might consider giving each of your clients a name

    with Client(s['address'], name='test-foo') as client

and then seeing which one isn't getting cleaned up?  You can also try `distributed.client.default_client()`.  We clean things up in the `__exit__()` call though, so as long as you're using context managers or `@gen_cluster` everything *should* work fine.